compiler-plugin/src/main/kotlin/it/unibo/collektive/frontend/checkers/NoAlignInsideLoop.kt

```diff
@@ -15,38 +20,63 @@ import org.jetbrains.kotlin.fir.expressions.FirWhileLoop
  */
 object NoAlignInsideLoop : FirFunctionCallChecker(MppCheckerKind.Common) {
 
-    private val safeOperators = listOf("alignedOn", "align", "dealign")
+    private val safeOperators = listOf(
+        "it.unibo.collektive.aggregate.api.Aggregate.alignedOn",
+        "it.unibo.collektive.aggregate.api.Aggregate.align",
+        "it.unibo.collektive.aggregate.api.Aggregate.dealign"
+    )
 
-    private fun CheckerContext.isInsideALoopWithoutAlignedOn(): Boolean {
-        val loopElementIndex = containingElements.indexOfLast { it is FirWhileLoop }
-        val functionDeclarationIndex = containingElements.dropLast(1).indexOfLast {
-            (it as? FirSimpleFunction)?.receiverParameter?.isAggregate(session) == true
+    /**
+     * Methods used inside collections to iterate their elements.
+     */
+    private val collectionMembers = listOf(
+        Class.forName("kotlin.collections.CollectionsKt"),
```

Comment:
Ouch, we should switch to the Kotlin reflection here. It may cause problems with phantom interfaces (such as `List`), but we can likely work them around

Reply:
I'm sorry, I'm a little confused. Should I revert the change to Java reflection? I converted it because it caused issues with the CI.

Reply:
Which issues? If there are issues in CI, there must be issues locally, too, or something's broken in CI.

There's no way Kotlin's reflection does not work in CI but works locally - right?
@nicolasfara

Reply:
@DanySK, is it possible to get the `CollectionsKt` methods via Kotlin reflection? I can't find another way to do this.

Reply:
@DanySK  What I meant was that the CI wasn't passing, because of some operation done by using Kotlin reflection. As stated by @nicolasfara, it's about the `CollectionsKt` file.

I've reverted the commit where I've switched to java reflection.

Reply:
@FreshMag so in general it was testing that was failing, everywhere, not just in CI, correct?

Reply:
The tests ran successfully on my local machine without any issues. I only noticed the problem with Kotlin reflection after the CI run failed.
[Here is the failed run](https://github.com/FreshMag/collektive-plugin/actions/runs/11541727746/job/32124073493?pr=12).

Reply:
@FreshMag, I've checked locally to see if the CI error is reproducible, and it is.
Could you test locally if the `./gradlew check --parallel` raises the same error as in CI?

Reply:
Surprisingly (at least for me), the `./gradlew dsl:compileKotlinJvm` succeeded, while `./gradlew compileKotlinLinuxX64` fails with the error reported above.

This is quite surprising since the compiler works on the JVM side, and should be independent from the target...
Taking a look at the exception, it seems that the Kotlin JVM reflection is used, not the target-specific one.

@DanySK any ideas?

Reply:
> @FreshMag, I've checked locally to see if the CI error is reproducible, and it is. Could you test locally if the `./gradlew check --parallel` raises the same error as in CI?

Yes, that command raised the same error locally on my machine.

Reply:
`Packages and file facades are not yet supported in Kotlin reflection. Meanwhile please use Java reflection to inspect this class: class kotlin.collections.CollectionsKt`

Nothing we can do: https://youtrack.jetbrains.com/issue/KT-16479

Let us just upvote on JetBrains' issue tracker, isolate the problem into a function, and write an implementation that relies on the Java reflection.

Reply:
And also leave a comment there linking the issue for future reference